# Dataverse - Bean and Brew - Sustainable Strategy

In [5]:
library(tidyverse)
#library(tidymodels)
#library(tidyclust)
library(repr)
library(GGally)
options(repr.matrix.max.rows = 6)
#source("cleanup.R")

In [28]:
items <- read_csv("items.csv")
items

Warning message:
“One or more parsing issues, call `problems()` on your data frame for details,
e.g.:
  dat <- vroom(...)
  problems(dat)”
Rows: 17 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): item_name, item_type, drink_temperature, drink_type
dbl (2): price, production_cost

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


item_name           price production_cost item_type   drink_temperature
1  Espresso             3.00 0.80            Drink       Hot              
2  Americano            3.50 1.00            Drink       Hot              
3  Latte                4.50 1.20            Drink       Hot              
4  Cappuccino           4.25 1.15            Drink       Hot              
5  Flat White           4.75 1.25            Drink       Hot              
6  Mocha                4.80 1.40            Drink       Hot              
7  Iced Coffee          4.00 1.10            Drink       Cold             
8  Cold Brew            4.75 1.35            Drink       Cold             
9  Iced Latte           4.60 1.25            Drink       Cold             
10 Iced Matcha Latte    5.25 1.60            Drink       Cold             
11 Matcha Latte         5.25 1.60            Drink       Hot              
12 Chai Latte           4.90 1.50            Drink       Hot              
13 Green Tea            2.90 0.20            Drink       Hot              
14 Hot Chocolate        4.25 1.10            Drink       Hot              
15 Reusable Coffee Cup 15.00 6.00            Merchandise N/A              
16 Tote Bag            12.00 4.00            Merchandise N/A              
17 Coffee Beans (1 lb) 18.00 8.00            Merchandise N/A              
   drink_type
1  Coffee    
2  Coffee    
3  Coffee    
4  Coffee    
5  Coffee    
6  Coffee    
7  Coffee    
8  Coffee    
9  Coffee    
10 Tea       
11 Tea       
12 Tea       
13 Tea       
14 Other     
15 NA        
16 NA        
17 NA

In [29]:
sales <- read_csv("sales.csv")
sales

Rows: 32605 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (3): item_name, transaction_type, own_cup
dbl  (1): customer_id
lgl  (1): surcharge
date (1): date
time (1): time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


date       time     item_name         transaction_type own_cup surcharge
1     2022-01-01 07:03:30 Cold Brew         Takeout          True    FALSE    
2     2022-01-01 07:30:58 Iced Coffee       Takeout          True    FALSE    
3     2022-01-01 08:32:23 Iced Latte        Takeout          False   FALSE    
4     2022-01-01 08:45:03 Latte             Dine-in          True    FALSE    
5     2022-01-01 09:03:47 Latte             Takeout          False   FALSE    
6     2022-01-01 10:02:30 Flat White        Takeout          True    FALSE    
7     2022-01-01 10:29:29 Iced Latte        Takeout          False   FALSE    
8     2022-01-01 10:53:39 Flat White        Takeout          True    FALSE    
9     2022-01-01 10:59:50 Americano         Dine-in          True    FALSE    
10    2022-01-01 11:02:54 Chai Latte        Takeout          False   FALSE    
11    2022-01-01 11:10:10 Iced Coffee       Takeout          True    FALSE    
12    2022-01-01 11:37:42 Matcha Latte      Takeout          False   FALSE    
13    2022-01-01 11:51:50 Cold Brew         Takeout          True    FALSE    
14    2022-01-01 12:35:16 Iced Latte        Takeout          True    FALSE    
15    2022-01-01 14:04:54 Latte             Takeout          True    FALSE    
16    2022-01-01 15:30:23 Cold Brew         Dine-in          True    FALSE    
17    2022-01-01 15:55:21 Iced Coffee       Takeout          True    FALSE    
18    2022-01-02 07:09:02 Iced Matcha Latte Takeout          False   FALSE    
19    2022-01-02 07:30:08 Matcha Latte      Dine-in          True    FALSE    
20    2022-01-02 07:40:06 Latte             Takeout          False   FALSE    
21    2022-01-02 07:56:27 Green Tea         Takeout          False   FALSE    
22    2022-01-02 08:26:25 Hot Chocolate     Takeout          True    FALSE    
23    2022-01-02 08:47:16 Iced Matcha Latte Takeout          False   FALSE    
24    2022-01-02 09:13:31 Iced Latte        Takeout          False   FALSE    
25    2022-01-02 09:27:33 Iced Latte        Takeout          False   FALSE    
26    2022-01-02 09:31:35 Latte             Takeout          True    FALSE    
27    2022-01-02 10:08:01 Chai Latte        Takeout          False   FALSE    
28    2022-01-02 10:19:33 Cold Brew         Dine-in          False   FALSE    
29    2022-01-02 10:25:25 Americano         Takeout          False   FALSE    
30    2022-01-02 10:30:10 Latte             Takeout          False   FALSE    
⋮     ⋮          ⋮        ⋮                 ⋮                ⋮       ⋮        
32576 2024-12-30 14:42:00 Matcha Latte      Dine-in          True    FALSE    
32577 2024-12-30 14:48:11 Americano         Takeout          False    TRUE    
32578 2024-12-30 15:04:51 Espresso          Takeout          True    FALSE    
32579 2024-12-30 15:19:34 Cappuccino        Takeout          True    FALSE    
32580 2024-12-30 15:26:00 Americano         Takeout          False    TRUE    
32581 2024-12-30 15:29:33 Cold Brew         Takeout          True    FALSE    
32582 2024-12-30 15:42:28 Americano         Takeout          False    TRUE    
32583 2024-12-31 07:07:32 Latte             Takeout          False    TRUE    
32584 2024-12-31 07:21:12 Iced Latte        Takeout          True    FALSE    
32585 2024-12-31 07:45:00 Iced Matcha Latte Dine-in          False   FALSE    
32586 2024-12-31 07:45:09 Iced Latte        Takeout          False    TRUE    
32587 2024-12-31 08:16:14 Iced Matcha Latte Takeout          True    FALSE    
32588 2024-12-31 08:23:06 Espresso          Dine-in          True    FALSE    
32589 2024-12-31 10:01:42 Americano         Takeout          False    TRUE    
32590 2024-12-31 10:44:21 Mocha             Dine-in          False   FALSE    
32591 2024-12-31 10:50:03 Matcha Latte      Takeout          False    TRUE    
32592 2024-12-31 11:15:27 Cold Brew         Dine-in          False   FALSE    
32593 2024-12-31 11:20:46 Mocha             Dine-in          True    FALSE    
32594 2024-12-31 11:33:29 Matcha Latte      Dine-in     

In [30]:
coffee <- inner_join(items,sales)
coffee

Joining with `by = join_by(item_name)`


item_name           price production_cost item_type   drink_temperature
1     Espresso            3     0.8             Drink       Hot              
2     Espresso            3     0.8             Drink       Hot              
3     Espresso            3     0.8             Drink       Hot              
4     Espresso            3     0.8             Drink       Hot              
5     Espresso            3     0.8             Drink       Hot              
6     Espresso            3     0.8             Drink       Hot              
7     Espresso            3     0.8             Drink       Hot              
8     Espresso            3     0.8             Drink       Hot              
9     Espresso            3     0.8             Drink       Hot              
10    Espresso            3     0.8             Drink       Hot              
11    Espresso            3     0.8             Drink       Hot              
12    Espresso            3     0.8             Drink       Hot              
13    Espresso            3     0.8             Drink       Hot              
14    Espresso            3     0.8             Drink       Hot              
15    Espresso            3     0.8             Drink       Hot              
16    Espresso            3     0.8             Drink       Hot              
17    Espresso            3     0.8             Drink       Hot              
18    Espresso            3     0.8             Drink       Hot              
19    Espresso            3     0.8             Drink       Hot              
20    Espresso            3     0.8             Drink       Hot              
21    Espresso            3     0.8             Drink       Hot              
22    Espresso            3     0.8             Drink       Hot              
23    Espresso            3     0.8             Drink       Hot              
24    Espresso            3     0.8             Drink       Hot              
25    Espresso            3     0.8             Drink       Hot              
26    Espresso            3     0.8             Drink       Hot              
27    Espresso            3     0.8             Drink       Hot              
28    Espresso            3     0.8             Drink       Hot              
29    Espresso            3     0.8             Drink       Hot              
30    Espresso            3     0.8             Drink       Hot              
⋮     ⋮                   ⋮     ⋮               ⋮           ⋮                
32576 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32577 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32578 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32579 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32580 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32581 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32582 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32583 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32584 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32585 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32586 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32587 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32588 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32589 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32590 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32591 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32592 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32593 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32594 Coffee Beans (1 lb) 18    8               Merchandise N/A              
32595 Coffee Beans (1 lb) 18

In [31]:
merch <- filter(coffee, item_type == "Merchandise")
merch

item_name           price production_cost item_type   drink_temperature
1    Reusable Coffee Cup 15    6               Merchandise N/A              
2    Reusable Coffee Cup 15    6               Merchandise N/A              
3    Reusable Coffee Cup 15    6               Merchandise N/A              
4    Reusable Coffee Cup 15    6               Merchandise N/A              
5    Reusable Coffee Cup 15    6               Merchandise N/A              
6    Reusable Coffee Cup 15    6               Merchandise N/A              
7    Reusable Coffee Cup 15    6               Merchandise N/A              
8    Reusable Coffee Cup 15    6               Merchandise N/A              
9    Reusable Coffee Cup 15    6               Merchandise N/A              
10   Reusable Coffee Cup 15    6               Merchandise N/A              
11   Reusable Coffee Cup 15    6               Merchandise N/A              
12   Reusable Coffee Cup 15    6               Merchandise N/A              
13   Reusable Coffee Cup 15    6               Merchandise N/A              
14   Reusable Coffee Cup 15    6               Merchandise N/A              
15   Reusable Coffee Cup 15    6               Merchandise N/A              
16   Reusable Coffee Cup 15    6               Merchandise N/A              
17   Reusable Coffee Cup 15    6               Merchandise N/A              
18   Reusable Coffee Cup 15    6               Merchandise N/A              
19   Reusable Coffee Cup 15    6               Merchandise N/A              
20   Reusable Coffee Cup 15    6               Merchandise N/A              
21   Reusable Coffee Cup 15    6               Merchandise N/A              
22   Reusable Coffee Cup 15    6               Merchandise N/A              
23   Reusable Coffee Cup 15    6               Merchandise N/A              
24   Reusable Coffee Cup 15    6               Merchandise N/A              
25   Reusable Coffee Cup 15    6               Merchandise N/A              
26   Reusable Coffee Cup 15    6               Merchandise N/A              
27   Reusable Coffee Cup 15    6               Merchandise N/A              
28   Reusable Coffee Cup 15    6               Merchandise N/A              
29   Reusable Coffee Cup 15    6               Merchandise N/A              
30   Reusable Coffee Cup 15    6               Merchandise N/A              
⋮    ⋮                   ⋮     ⋮               ⋮           ⋮                
2115 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2116 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2117 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2118 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2119 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2120 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2121 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2122 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2123 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2124 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2125 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2126 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2127 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2128 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2129 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2130 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2131 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2132 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2133 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2134 Coffee Beans (1 lb) 18    8               Merchandise N/A              
2

In [10]:
coffee_only <- coffee |> 
  filter(item_type != "Merchandise") |>
  mutate(
    drink_temperature = as_factor(drink_temperature),
    drink_type        = as_factor(drink_type),
    transaction_type  = as_factor(transaction_type),
    own_cup           = as.logical(own_cup)
  )
coffee_only

item_name     price production_cost item_type drink_temperature
1     Espresso      3     0.8             Drink     Hot              
2     Espresso      3     0.8             Drink     Hot              
3     Espresso      3     0.8             Drink     Hot              
⋮     ⋮             ⋮     ⋮               ⋮         ⋮                
30459 Hot Chocolate 4.25  1.1             Drink     Hot              
30460 Hot Chocolate 4.25  1.1             Drink     Hot              
30461 Hot Chocolate 4.25  1.1             Drink     Hot              
      drink_type date       time     transaction_type own_cup surcharge
1     Coffee     2022-01-02 10:43:09 Takeout          FALSE   FALSE    
2     Coffee     2022-01-02 15:26:48 Takeout           TRUE   FALSE    
3     Coffee     2022-01-03 07:18:28 Dine-in          FALSE   FALSE    
⋮     ⋮          ⋮          ⋮        ⋮                ⋮       ⋮        
30459 Other      2024-12-29 11:37:28 Dine-in          FALSE   FALSE    
30460 Other      2024-12-30 07:47:35 Takeout           TRUE   FALSE    
30461 Other      2024-12-31 13:15:02 Takeout           TRUE   FALSE    
      customer_id
1     18500      
2     11609      
3     10437      
⋮     ⋮          
30459 25413      
30460 11586      
30461 11609

In [11]:
coffee_drink_type <- coffee_only|> 
  group_by(drink_temperature) |> 
  summarise(brought_cup = sum(own_cup == TRUE),not_brought_cups = sum(own_cup == FALSE ),proportion = mean(own_cup == TRUE), proportion_not_own_cup = mean(own_cup == FALSE))
coffee_drink_type

drink_temperature brought_cup not_brought_cups proportion
1 Hot               10509       10109            0.5097003 
2 Cold               2758        7085            0.2801991 
  proportion_not_own_cup
1 0.4902997             
2 0.7198009

##### Explanation of how drink_temperature could affect reusability of cups
The data suggests that hot drinks are much more likely to be ordered using a reusable cup than cold drinks. The proportion of customers bringing their own cup for hot beverages is around 51%, whereas for cold beverages it drops to about 28%. This pattern can be interpreted in several realistic ways:
###### Practicality and Habit
- Customers tend to associate reusable cups (e.g., travel mugs, thermoses) with hot drinks like coffee or tea. Many people already own insulated hot drink tumblers and routinely carry them, especially commuters. Therefore, they naturally use them when purchasing hot drinks.
- Cold beverages, however, are more often consumed on the go without preplanning, meaning customers are less likely to bring a cup specifically for a cold drink.
###### Type of Container Used
- Reusable hot drink mugs are widely available, standardized, and easy for shops to fill.
- Reusable containers for cold drinks (e.g., clear tumblers for iced drinks) are less commonly owned, and many customers may not think of bringing one.
- Some coffee shops also have stricter guidelines for filling customer-provided containers for iced/blended drinks, reducing incentive.
###### Seasonal/Comfort Motivation
- Hot drinks are often purchased during colder weather, when customers already carry bags, backpacks, or jackets—making it easy to carry a reusable cup.
- Cold drinks are often spontaneous purchases in warm weather (e.g., “I’m hot, I’ll grab an iced coffee”), so people may not have brought a reusable container with them.

###### Conclusion
Overall, the data indicates that hot drink customers engage in sustainable behavior more consistently, likely due to established habits, availability of suitable reusable containers, and the practical advantages of insulated travel mugs. Cold drink consumers show much lower reusability, suggesting that encouraging reusable cold drink tumblers (or offering incentives) could significantly reduce waste in that category.